### PR TITLE
Check for null TtlAction in DefaultFileSystemMaster

### PR DESCRIPTION
**Pull request by ZacBlanco**
TtlAction needs to check for null in DefaultFIleSystemMaster

The setAttribute function should only change the TtlAction if it's specifically set within the options. Otherwise, we should avoid the set. Setting the TtlAction with a null param results in the Ttl Action being set to DELETE. If the default is FREE, then a user may be inadvertently setting the action to DELETE by calling SetAttribute.
    
A test is added which fails before this fix.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -3607,7 +3607,7 @@ private void setAttributeSingleFile(RpcContext rpcContext, LockedInodePath inode
       if (ttl != null && inode.getTtl() != ttl) {
         entry.setTtl(ttl);
       }
-      if (inode.getTtlAction() != action) {
+      if (action != null && inode.getTtlAction() != action) {
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: I think the check on line 3614 should also have the other two conditions (`get* != *`)?

Comment (ZacBlanco) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: The last mod time should only ever be updated if the entry was actually modified. That behavior is consistent with the way `ext4` behaves on my CentOS 6 machine. Pushed a fix in f2a5fcf

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 2]
```diff
@@ -3603,15 +3603,18 @@ private void setAttributeSingleFile(RpcContext rpcContext, LockedInodePath inode
...
         entry.setTtlAction(ProtobufUtils.toProtobuf(action));
+        modified = true;
       }
-      // We've set at least one if they are both non-null
-      if (ttl != null && action != null) {
+
+      if (modified) {
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: Could you cover this in a test?

Comment (ZacBlanco) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: done

**Comment by ZacBlanco** [comment]
jenkins, test this please

**Pull request merged**